test/Microsoft.AspNetCore.Http.Connections.Tests/HttpConnectionDispatcherTests.cs

```diff
@@ -338,6 +338,59 @@ public HttpConnectionDispatcherTests(ITestOutputHelper output) : base(output)
             }
         }
 
+        [Theory]
+        [InlineData(HttpTransportType.LongPolling)]
+        [InlineData(HttpTransportType.ServerSentEvents)]
+        public async Task PostSendsToConnectionInParallel(HttpTransportType transportType)
+        {
+            using (StartLog(out var loggerFactory, LogLevel.Debug))
+            {
+                var manager = CreateConnectionManager(loggerFactory);
+                var dispatcher = new HttpConnectionDispatcher(manager, loggerFactory);
+                var connection = manager.CreateConnection();
+                connection.Items[ConnectionMetadataNames.Transport] = transportType;
+
+                List<Task> sendTasks = new List<Task>();
+                // This tests thread safety of sending multiple pieces of data to a connection at once
+                for (int i = 0; i < 1000; i++)
```

Comment:
We try not to test race conditions this way.

If you can't find a way to block inside the lock via a mock or something then I don't think we should have a test for this

Reply:
Ok, I'm not a big fan of brute forcing concurrency issues either.

I'll see if I can come up with something more elegant.